### DISCUSSION TITLE
How to change Google-Mobile-Ads-SDK version

### DISCUSSION THREAD

**Discussion by terfender** [invertase/react-native-firebase#5214] [Q&A]
How can I change `@react-native-firebase/admob` to use `Google-Mobile-Ads-SDK` version `8.1.0`?
I am not sure where `7.66` is set. Can you help me please?

```
[!] CocoaPods could not find compatible versions for pod "Google-Mobile-Ads-SDK":
  In Podfile:
    Google-Mobile-Ads-SDK (~> 8.1.0)

    RNFBAdMob (from `../node_modules/@react-native-firebase/admob`) was resolved to 11.3.3, which depends on
      Firebase/AdMob (= 7.10.0) was resolved to 7.10.0, which depends on
        Google-Mobile-Ads-SDK (~> 7.66)

```

Thanks

**Comment by mikehardy** [comment]
It's set by this chain of dependencies:
- react-native-firebase/admob has a reference to the firebase-ios-sdk via it's .podspec file
- that reference consumes the react-native-firebase/app firebase-ios-sdk version from it's package.json file
- the firebase-ios-sdk podspec itself defines all of it's dependencies, including for the Ads SDKs

I believe if you examine Podfile.lock you'll see these things.
I don't believe that firebase is compatible with the higher versions of the Ads SDK unless it says so, and that's a breaking change version bump (from 7 major to 8 major in semantic versioning terms) so I doubt it will work

You should examine the firebase-ios-sdk github repo to see if they have plans to be compatible - but also note that we don't support the new version of firebase-android-sdk (v27) because it uses a new breaking-change version of android AdMob and we're not compatible here yet #5150 

Sometimes when you are working with a large / complicated system like firebase which has lots of transitive dependencies (like admob - just one dependency among quite a few) - ingesting changes in the transitive dependencies takes some time
